src/cloudevent/builder/ucloudeventutils.rs

```diff
@@ -144,7 +144,9 @@ impl UCloudEventUtils {
     /// otherwise a `None` is returned.
     pub fn get_ttl(event: &Event) -> Option<u32> {
         if let Some(ExtensionValue::Integer(ttl)) = event.extension("ttl") {
-            return Some(*ttl as u32);
+            if let Ok(ttl) = u32::try_from(*ttl) {
```

Comment:
Converting a `Result<T, E>` into an `Option<T>` can be easily done with the `ok()` function. See [here](https://doc.rust-lang.org/std/result/enum.Result.html#method.ok).

Reply:
done